common/schema-builder/src/main/java/io/smallrye/graphql/schema/creator/ReferenceCreator.java

```diff
@@ -390,6 +390,14 @@ private void putIfAbsent(String key, Reference reference, ReferenceType referenc
             if (!map.containsKey(key)) {
                 map.put(key, reference);
                 queue.add(reference);
+            } else {
+                String existingClass = map.get(key).getClassName();
+                String newClass = reference.getClassName();
+                if (!existingClass.equals(newClass)) {
+                    throw new SchemaBuilderException(
+                            "Classes " + existingClass + " and " + newClass + " map to the same GraphQL type '" + key + "', " +
+                                    "consider using the @Name annotation to distinguish between them");
+                }
```

Comment:
Maybe also mention the naming strategy in the error message. I was also wondering if we should fail, or just repeat the name and count. Or add a config that only do the naming strategy on clash. Then add a nameing strategy that count

Reply:
By counting you mean appending a number to the name? That doesn't sound like a good idea, numbers aren't a good way to distinguish the types, and we would have to employ some mechanisms to make sure the numbering is predictable (how to decide which type should be 1 and which should be 2?)
Anyway I changed the message to suggest using a different naming strategy

Reply:
The model already have the class name, but agree, let's leave it like this for now